java/gradle.java/src/org/netbeans/modules/gradle/java/classpath/GradleSourcesImpl.java

```diff
@@ -159,7 +160,7 @@ public synchronized SourceGroup[] getSourceGroups(String type) {
         for (SourceType st : stype) {
             Set<File> processed = new HashSet<>();
             for (String group : gradleSources.keySet()) {
-                Set<File> dirs = gradleSources.get(group).getSourceDirs(st);
+                Set<File> dirs = filterEmptyGeneratedDirs(gradleSources.get(group).getSourceDirs(st));
```

Comment:
Is it OK to filter the empty dirs from the data layer ? I'd rather leave them discoverable (as they are valid source roots), but filter them from the UI only.

Reply:
Good insight! Thanks! Moving it to the UI!